docs/ARCHITECTURE_DOCUMENT.md

```diff
@@ -0,0 +1,258 @@
+
+### 2.1 Model
+
+É uma representação do banco de dados. Além disso, também inclui características, relações e outros comportamentos que os dados podem assumir.
+Para adicionar os valores no banco, é necessário apenas  defini-los de acordo com suas variáveis correspondentes e o django em si, comunica-se com o banco de forma inteligente.
+
+### 2.2 View
+
+Estabelece uma ponte entre a Models e o Templates. Recebe os requests do usuário a partir do template, acessa o banco de dados e então retorna informação requerida pelo usuário, como HTML, XML e erros encontrados.
+
+### 2.3 Template
+
+Agrega toda a parte visual que estará visível para os usuários. Inclui os códigos HTML, CSS, javascript, entre outras linguagens que concentram na apresentação da View ao usuário.
+
+## 3:  Organização Arquitetural
+
+### 3.1  Class-Based Views
+
+Proporcionam um método alternativo para implementar views como objetos ao em vez de funções. As CBV são classes que implementam métodos e atributos que são comumente utilizados na programação das views. Dessa maneira, o programador pode utilizar métodos já implementados ou subscrever esses métodos e implementar da sua maneira. Para atender os mais variados casos de uso das views, as CBV oferecem diversos temas para implementação.
+
+## 4:  Requisitos e Restrições Arquiteturais
+
+Linguagem	Python 3.6.4
+
+Framework	Django  2.0.3
+
+Plataforma	Web - Navegadores Google Chrome, Safari e Firefox
+
+Segurança	O sistema terá informações pessoais dos pacientes que só poderão ser vistas pelo mesmo ou pelo(s) seu(s) respectivo(s) médico(s). Outros dados pessoais só poderão ser vistos pelo  usuário a quem esses dados se referem.
+
+Internacionalização (i18n)	A aplicação terá suporte para aos idiomas: Inglês e Português (linguagem padrão).
+
+## 5:	Visão Lógica
+
+### 5.1	Visão Geral: Pacotes e Camadas
+
+O framework Django organiza os projetos em apps, que são pastas que contém uma funcionalidade independente do restante da aplicação. Além disso, existem arquivos de configuração e arquivos estáticos globais. A figura a seguir mostra a organização de pastas de um app.
+
+## 6:	Visão de Implementação
+
+### 6.1 Diagrama de Pacotes
+
+![Diagrama de Pacotes](http://uploaddeimagens.com.br/images/001/350/330/full/DP.png?1522284479)
+
+### 6.2 Diagrama de Classes
+
+![Diagrama de Classes](http://uploaddeimagens.com.br/images/001/350/293/full/DC.png?1522283843)
+
+### 6.3 Modelo Entidade Relacionamento (MER)
+
+### USER:
+
+| Atributo | Tipo | Característica | Descrição |
+|  ---  |  ---  |  ---  |  ---  |
+| name | string[200] | obrigatório | Nome completo do usuário |
+| email | string[50] | obrigatório, único | Email será usado como username do usuário |
+| telephone| string[14] | obrigatório | Telefone do usuário |
```

Comment:
telefone não é string, número de celular não tem letra.
@VictorArnaud favor opine neste ponto

Reply:
e se for +5561999999999 ?

Reply:
Acho que aqui é algo parecido com o cpf. O @VictorArnaud tem que avaliar se é melhor salvar pré-formatado para exibição (talvez através dos validadores de campo) ou salvar uma int.

Reply:
normalmente a gente salva sem formatar e formata só na exibição para o usuário